File: 2025_Oct/Kyle/2025-10-27-Longest-Repeating-Character-Replacement.py
class Solution(object):
    def characterReplacement(self, s, k):
        from collections import Counter

        left = right = 0
        max_length = 0
        while right < len(s):
            for i in range(left, len(s)):
                while left <= right:
                    counts = Counter()
                    for char in s[i:right]:
                        counts[ord(char) - ord('A')] += 1
                    max_count = max(counts.values() or [0])

                if len(s[i:right]) - max_count <= k:
                    max_length = max(max_length, len(s[i:right]))
                right += 1
            left += 1

        return max_length

File: 2025_Oct/youngwon/2025-10-27-Longest-Repeating-Character-Replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> count(26, 0);
        int maxCount = 0;
        int left = 0, result = 0;

        for (int right = 0; right < s.size(); right++) {
            count[s[right] - 'A']++;
            maxCount = max(maxCount, count[s[right] - 'A']);

            if (right - left + 1 - maxCount > k) {
                count[s[left] - 'A']--;
                left++;
            }

            result = max(result, right - left + 1);
        }

        return result;
    }
};
